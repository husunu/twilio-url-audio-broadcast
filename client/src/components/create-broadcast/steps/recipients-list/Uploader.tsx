import React, { memo } from 'react';
import { Typography, Box, Button } from '@material-ui/core';

import { useUploaderStyles } from './styles';
import { DropzoneInputProps, DropzoneRootProps } from 'react-dropzone';

type UploaderProps = {
  getRootProps(): DropzoneRootProps;
  getInputProps(): DropzoneInputProps;
  selectedFile: File | null;
};

const Uploader: React.FC<UploaderProps> = memo(
  ({ getRootProps, getInputProps, selectedFile }) => {
    const classes = useUploaderStyles();

    return (
      <Box
        data-testid="drop-zone"
        {...getRootProps()}
        className={classes.uploader}
      >
        <input {...getInputProps()} />
        <Box className={classes.uploaderItems}>
          <Typography variant="h5" align="center">
            Drag & Drop Recipient File (.txt)
          </Typography>
        </Box>
        <Box className={classes.uploaderItems}>
          <Typography variant="h6" align="center">
            or
          </Typography>
        </Box>
        <Box className={classes.uploaderItems}>
          <Button variant="contained" color="primary">
            {selectedFile ? 'Select New File' : 'Select File'}
          </Button>
        </Box>
        <Box className={classes.uploaderItems}>
          {selectedFile && <Typography>{selectedFile.name}</Typography>}
        </Box>
      </Box>
    );
  }
);

export default Uploader;
