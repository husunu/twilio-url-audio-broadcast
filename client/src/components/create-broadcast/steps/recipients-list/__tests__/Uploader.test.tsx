import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import { useDropzone } from 'react-dropzone';
import Uploader from '../Uploader';

const TestUploader = () => {
  const { getRootProps, getInputProps } = useDropzone();

  return (
    <Uploader
      getInputProps={getInputProps}
      getRootProps={getRootProps}
      selectedFile={null}
    />
  );
};

describe('recipients list > Uploader', () => {
  test('renders the page', () => {
    const { asFragment } = render(
      <Router>
        <TestUploader/>
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
