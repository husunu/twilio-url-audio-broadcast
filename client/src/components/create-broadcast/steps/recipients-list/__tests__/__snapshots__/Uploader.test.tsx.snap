// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipients list > Uploader renders the page 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-uploader-1"
    data-testid="drop-zone"
    tabindex="0"
  >
    <input
      autocomplete="off"
      multiple=""
      style="display: none;"
      tabindex="-1"
      type="file"
    />
    <div
      class="MuiBox-root MuiBox-root-5 makeStyles-uploaderItems-2"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
      >
        Drag & Drop Recipient File (.txt)
      </h5>
    </div>
    <div
      class="MuiBox-root MuiBox-root-36 makeStyles-uploaderItems-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
      >
        or
      </h6>
    </div>
    <div
      class="MuiBox-root MuiBox-root-37 makeStyles-uploaderItems-2"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Select File
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root MuiBox-root-70 makeStyles-uploaderItems-2"
    />
  </div>
</DocumentFragment>
`;
